### ISSUE TITLE
Create SIF Folder Refund Process & Features (LP) (PROD UAT In Progress TL)

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#11875]
### Unable to process SIF refund. No refund doc. Getting error message saying 'No Valid Refund

- Update payment type column in valid folder to Check
- Add document in the tab mask
- Create new doc 'SIF-Request for Refund' (copy template from RW-Request for Refund)
- Default insert refund document at folder creation

![image](https://user-images.githubusercontent.com/49930868/228672615-77cc733f-3903-4c05-bb66-49f85e74adf7.png)

**Comment by TracyLinder** [comment]
Tracy needs permission to refund in test

**Comment by TracyLinder** [comment]
### Sync / Planning
_Tracy, Priya, Vish_

- Updated AC w/ list of things Priya needs to do to get refunds working on SIF
- Put on deck for Priya
- Asked to get this in PROD by 4/21/23
- Tracy can do the UAT
- Vish will be available to Priya if help is needed

**Comment by TracyLinder** [comment]
@Lpriya-LIT can you please add a status update. Thank you.

**Comment by Lpriya-LIT** [comment]
@Nadin-Nader Committed changes and requested for Dev and Test Promotion.

**Comment by Nadin-Nader** [comment]
@Lpriya-LIT  Can you put the Service Now task number in a comment please?

**Comment by Lpriya-LIT** [comment]
@Nadin-Nader Devops commit has been promoted to Test.
Document needs to be deployed manually, working with Amanda Support team.

**Comment by Lpriya-LIT** [comment]
@Nadin-Nader Changes deployed to Test , getting a different error when trying to do refund . Analysis in Progress

**Comment by Lpriya-LIT** [comment]
@Nadin-Nader Committed changes , ready for deployment SCTASK0309132

**Comment by TracyLinder** [comment]
### Update from Nathan (Refund request for SIF 
- The refund has been issued. ATD finance Adelita did the refund manually. 
- We will need to add the refund to the SIF folder after this is fixed

**Comment by Lpriya-LIT** [comment]
@Nadin-Nader Ready for UAT on Test

**Comment by TracyLinder** [comment]
### UAT Results

I created test folder 12892007
I paid online
I adjusted the bill
❌ Tried to apply refund, but I don't have permissions to process credit card refunds. I'll check w/ Nate to get permissions. 
I created a new bill, paid it with a check, and was able to process the refund. 











**Misc. Note:**

1. Can't finish UAT until a credit card payment refund test is done
2. Question for TDS: What refund reason will be used? Does it already exist or do we need to create one for just SIF?
3. SIF folders do not appear on portal in the **My Permits/Cases** (Is this a bug? Has it ever shown here?)

**Comment by TracyLinder** [comment]
### UAT Update
- There may be a permissions bug because i should be able to refund credit cards with my test permissions. 
- We're working w/ Crystal T. in finance to have her test a credit card refune.

**Comment by TracyLinder** [comment]
### UAT Update - Approved for PROD ✅
FYI @Lpriya-LIT 

- Crystal was able to process the refund for the credit card payment. 
    - ROWID 12892012, Payment ID 7445893

 
<img width="1170" alt="Screenshot 2023-05-23 at 4 54 30 PM" src="https://github.com/cityofaustin/atd-data-tech/assets/49930868/a199c430-25a0-495e-900a-f553281c7590">

**Comment by TracyLinder** [comment]
### Refund Document Bug

- The refund doc shows all adjusted fees instead of the 2 specific fees i was testing in my scenario. 
    - ROWID 12892012
    - DOCRSN 27686543
    - Fees being refunded are in the screenshot
    - The doc is in the next screenshot

![image](https://github.com/cityofaustin/atd-data-tech/assets/49930868/85cf73e9-3df5-4cd4-ab17-3ce33fab4148)

![image](https://github.com/cityofaustin/atd-data-tech/assets/49930868/05c175ec-e148-455e-9298-8ffcba179c3b)

**Comment by TracyLinder** [comment]
### Another doc example
- ROWID 12892141

![image](https://github.com/cityofaustin/atd-data-tech/assets/49930868/bc670883-999d-4770-8692-6437962035bb)

**Comment by TracyLinder** [comment]
### Promote to PROD
- Fixing the refund document is not in the scope of this request. We'll address the concern at another time.

**Comment by Lpriya-LIT** [comment]
@Nadin-Nader Created Prod promotion request - SCTASK0320920

**Comment by Nadin-Nader** [comment]
@Lpriya-LIT  This has been deployed to PROD I will send Kyle and ROW a message asking for them to UAT.

**Comment by Nadin-Nader** [comment]
Sent a TEAMS message to Amber asking her to UAT this on PROD. Waiting on Response

**Comment by Nadin-Nader** [comment]
Sent another TEAMS message to Amber, Nathan asking her to UAT this on PROD. Waiting on Response

**Comment by Nadin-Nader** [comment]
From Amber:

[Yesterday 12:43 PM] Hutchens, Amber
Hey Nadin - Sorry, I'm still catching up from my vacation last week. 

[Yesterday 12:43 PM] Hutchens, Amber
I'll get to this soon.

**Comment by TracyLinder** [comment]
### PROD UAT Scheduled
- Meeting with Nathan & Don on 7/18/23 to process a real refund in AMANDA.

**Comment by TracyLinder** [comment]
### PROD UAT UPDATE
- **TL;DR: We met for testing today and were able to generate the refund document, print it, and send it. Finance needs to do their portion.** 

**Followup Email:**

Hi Nate and Dan,

Attached is the current AMANDA refund training doc. According to the doc you are almost done with the refund. You still need to send finance the refund request documents and be sure to note the refund has already been processed. (Start on page 8.) Can you please copy me on the finance email in case they encounter a bug on their side? 

List of Asks:
Finish the refund; start on page 8. 
Copy me on the email to finance. 

Thank You,

**Comment by TracyLinder** [comment]
The end user was able to generate the documents and send them to finance today. Now we wait for finance to confirm they were able to do their portion.

**Comment by TracyLinder** [comment]
I believe this to have been successful. Finance was able to process the refunds.
